src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1329,9 +1443,22 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Set(spinel_prop_key_t aK
     assert(mWaitingTid == 0);
 
     mExpectedCommand = SPINEL_CMD_PROP_VALUE_IS;
+    mRcpRecoveredInOperation = false;
+
     va_start(mPropertyArgs, aFormat);
     error = RequestV(true, SPINEL_CMD_PROP_VALUE_SET, aKey, aFormat, mPropertyArgs);
     va_end(mPropertyArgs);
```

Comment:
If RCP failure happens during calling `RequestV`, then in the recovery process `Set` is called before `RequestV` returns, `mPropertyArgs` will be overwritten and cause a crash. `mExpectedCommand` will also be overwritten.

Maybe we need to make these functions reentrant.

Reply:
This issue was solved by avoiding reentrancy. Moved recovery process out of `RequestV` calls.